contracts/factory/src/storage.rs

```diff
@@ -1 +1,99 @@
+use crate::error::ContractError;
+use soroban_sdk::{
+    contracttype, symbol_short, Address, BytesN, ConversionError, Env, Symbol, TryFromVal, Val,
+};
 
+#[derive(Clone, Copy)]
+#[repr(u32)]
+pub enum DataKey {
+    Admin = 1,
+}
+
+impl TryFromVal<Env, DataKey> for Val {
+    type Error = ConversionError;
+
+    fn try_from_val(_env: &Env, v: &DataKey) -> Result<Self, Self::Error> {
+        Ok((*v as u32).into())
+    }
+}
+
+#[contracttype]
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct LiquidityPoolInitInfo {
+    admin: Address,
+    lp_wasm_hash: BytesN<32>,
+    share_token_decimals: u32,
+    swap_fee_bps: i64,
+    fee_recipient: Address,
+    max_allowed_slippage_bps: i64,
+    max_allowed_spread_bps: i64,
+}
+
+#[contracttype]
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct TokenInitInfo {
+    pub token_wasm_hash: BytesN<32>,
+    pub token_a: Address,
+    pub token_b: Address,
+}
+
+#[contracttype]
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct StakeInitInfo {
+    pub stake_wasm_hash: BytesN<32>,
+    pub min_bond: i128,
+    pub max_distributions: u32,
+    pub min_reward: i128,
+}
```

Comment:
Check out: https://github.com/Phoenix-Protocol-Group/phoenix-contracts/issues/113

Reply:
started the draft in a separate branch, will keep this here for now and once this is ready will refactor from [#113](https://github.com/Phoenix-Protocol-Group/phoenix-contracts/issues/113)